app/views/listings/_listing_actions.haml

```diff
@@ -23,61 +23,108 @@
+                    return false;
+                  }
+
+                  var startDate = startVal[0].getTime();
+                  var endDate = endVal[0].getTime();
+
+                  // Validate that all booked dates are outside the selected range
+                  return disabledDates.every(function(d) {
+                    var date = d.getTime();
+                    if (startDate === endDate) {
+                      return date !== startDate;
+                    }
+                    return date < startDate || date >= endDate;
+                  });
+                });
+              var rules = #{@listing.quantity_selector&.to_sym == :night} ? {
+                "end_on": {
+                  night_selected: {startOnSelector: "#start-on"},
+                  availability_range: {startOnSelector: "#start-on"}
+                }
+              } : {
+                "end_on": {
+                  availability_range: {startOnSelector: "#start-on"}
+                }
+              };
+              $("#booking-dates").validate({
+                rules: rules,
+                submitHandler: function(form) {
+                  var $form = $(form);
+                  $form.find("#start-on").attr("name", "");
+                  $form.find("#end-on").attr("name", "");
+
+                  form.submit();
                 }
               });
-            var rules = #{@listing.quantity_selector&.to_sym == :night} ? {
-              "end_on": {night_selected: {startOnSelector: "#start-on"}}
-            } : {};
-            $("#booking-dates").validate({
-              rules: rules,
-              submitHandler: function(form) {
-                var $form = $(form);
-                $form.find("#start-on").attr("name", "");
-                $form.find("#end-on").attr("name", "");
 
-                form.submit();
-              }
-            });
+              window.ST.initializeFromToDatePicker('datepicker', {disabledDates: disabledDates});
+          .input-daterange.input-group.clearfix#datepicker{:data => {:locale => I18n.locale, :dateformat => t("datepicker.format")}}
+            .datepicker-start-wrapper
+              = label_tag t(".booking_from")
+              %input.input-sm.form-control.required#start-on{:type => 'text', :name => "start_on", :placeholder => t("datepicker.format"), :data => { :output => "booking-start-output" } }
+              %input#booking-start-output{:type => 'hidden', :name => 'start_on'}
+
+            .datepicker-end-wrapper
+              = label_tag t(".booking_to")
+              %input.input-sm.form-control.required#end-on{:type => 'text', :name => "end_on", :placeholder => t("datepicker.format"), :data => { :output => "booking-end-output" }}
+              %input#booking-end-output{:type => 'hidden', :name => 'end_on'}
+        - blocked_dates_result.on_error do
+
+          .listings-availability-information-error
+            = "Could not load availabilty information. Please try again later."
```

Comment:
This error str is not hard coded. I guess there's no reason to have it localized?